doc/user/environments.xml

```diff
@@ -1538,6 +1548,79 @@ env.PrependUnique(CCFLAGS=['-g'])
 
       </para>
 
+      </section>
+
+      <section>
+      <title>Overriding Construction Variable Settings</title>
+
+      <para>
+
+      Rather than creating a cloned environmant for specific tasks,
+      you can <firstterm>override</firstterm> or add construction variables
+      when calling a builder method by passing them as keyword arguments.
+      The values of these overridden or added variables will only be in
+      effect when building that target, and will not affect other parts
+      of the build.
+      For example, if you want to add additional libraries for just one program:
+
+      </para>
+
+      <programlisting>
+env.Program('hello', 'hello.c', LIBS=['gl', 'glut'])
+      </programlisting>
+
+      <para>
+
+      or generate a shared library with a non-standard suffix:
+
+      </para>
+
+      <programlisting>
+env.SharedLibrary(
+    target='word',
+    source='word.cpp',
+    SHLIBSUFFIX='.ocx',
+    LIBSUFFIXES=['.ocx'],
+)
+      </programlisting>
+
+      <para>
+
+      When overriding this way, the Python-style keyword arguments
+      mean "set to this value".  If you want your override to augment
+      an existing value, you have to take some extra steps. XXX
+
+      </para>
+
+      <para>
+
+      It is also possible to use the <parameter>parse_flags</parameter>
+      keyword argument in an override to merge command-line
+      style arguments into the appropriate construction
+      variables. This works like the &f-link-env-MergeFlags; method,
+      which will be described later.
+
+      </para>
+
+      <para>
+
+      This example adds 'include' to &cv-link-CPPPATH;,
+      'EBUG' to &cv-link-CPPDEFINES;, and 'm' to &cv-link-LIBS;:
+
+      </para>
+
+      <programlisting>
+env = Program('hello', 'hello.c', parse_flags='-Iinclude -DEBUG -lm')
+      </programlisting>
+
+      <para>
+
+      Using temporary overrides this way is lighter weight than making
+      a full construction environment, so it can help performance in
+      large projects which have lots of special case values to set.
+
+      </para>
+
```

Comment:
Maybe a note about two builders using same sources but different overrides yielding error about two different environments for the same target?

Reply:
And you wanted to keep it simpler?  :)

Reply:
added.